apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/page/content/subsection/SubSectionCreator.java

```diff
@@ -37,6 +43,42 @@ public CreateSubSectionResponse createSubSection(long pageId, Long userId, Creat
         }
     }
 
+
+    public DeleteSubSectionResponse deleteSubSection(DeleteSubSectionRequest request) {
+        try {
+            log.info("Deleting section");
+            Integer order_nbr = waUiMetaDataRepository.getOrderNumber(request.subSectionId());
+            waUiMetaDataRepository.deletefromTable(request.subSectionId());
+            waUiMetaDataRepository.updateOrderNumberByDecreasing(order_nbr, request.subSectionId());
+            return new DeleteSubSectionResponse(request.subSectionId(), "Sub Section Deleted Successfully");
+        } catch(Exception exception) {
+            throw new DeleteSubSectionException("Delete Sub Section exception", 1015);
+        }
+
+    }
+
+    public UpdateSubSectionResponse updateSubSection(UpdateSubSectionRequest request) {
+        try {
+            log.info("Updating section");
+            if (request.questionLabel() != null && request.visible() != null) {
+                waUiMetaDataRepository.updateQuestionLabelAndVisibility(request.questionLabel(), request.visible(), request.subSectionId());
+                return new UpdateSubSectionResponse(request.subSectionId(), "SubSection Updated Successfully");
+            } else if ( request.questionLabel() != null ) {
+                waUiMetaDataRepository.updateQuestionLabel(request.questionLabel(), request.subSectionId());
+                return new UpdateSubSectionResponse(request.subSectionId(), "SubSection Updated Successfully");
+            } else if ( request.visible() != null ) {
+                waUiMetaDataRepository.updateVisibility(request.visible(), request.subSectionId());
+                return new UpdateSubSectionResponse(request.subSectionId(), "SubSection Updated Successfully");
+            } else {
+                return new UpdateSubSectionResponse(request.subSectionId(), "questionLabel or Visible is required to update subsection");
+            }
```

Comment:
I think we should require both the label and the `visible` values to be present. Then if either are null, throw an exception. Then this becomes something like:

```java
if (request.questionLabel() == null || request.visible() == null) {
    throw new UpdateSubSectionException("Label and visibility fields are required");
}
waUiMetaDataRepository.updateQuestionLabelAndVisibility(request.questionLabel(), request.visible(), request.subSectionId());
return new UpdateSubSectionResponse(request.subSectionId(), "SubSection Updated Successfully");
```